libsolidity/codegen/ArrayUtils.cpp

```diff
@@ -896,11 +896,16 @@ void ArrayUtils::popStorageArrayElement(ArrayType const& _type) const
 		// Stack: ArrayReference oldLength
 		m_context << u256(1) << Instruction::SWAP1 << Instruction::SUB;
 		// Stack ArrayReference newLength
-		m_context << Instruction::DUP2 << Instruction::DUP2;
-		// Stack ArrayReference newLength ArrayReference newLength;
-		accessIndex(_type, false);
-		// Stack: ArrayReference newLength storage_slot byte_offset
-		StorageItem(m_context, *_type.baseType()).setToZero(SourceLocation(), true);
+
+		if (_type.baseType()->category() != Type::Category::Mapping)
```

Comment:
I'm confused. Isn't this supposed to be `==` because the change log refers to array of mappings.

Reply:
Mappings can't be zeroed, so if the base type is a mapping nothing should happen. If it's not a mapping, the content should be zeroed.